File: DTE_Maker/DTE.cs
using DTE_Maker;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Serialization;

//felipe pasache 2018 07 23 a partir de entidades se crea el xml en base a la serializacion 
public class MakeDte : Extends
{
    [XmlRoot("DTE"), Serializable]
    public class DTE
    {
        [XmlAttribute("version")]
        public string version = "1.0";
        [XmlIgnore]
        public string TipoOperacion;
        [XmlElement("Documento")]
        public Documento documento = new Documento();
    }
    public class Documento
    {
        [XmlAttribute("ID")]
        public string ID { get; set; }

        [XmlElement("Encabezado")]
        public Encabezado encabezado = new Encabezado();
        [XmlElement("Detalle")]
        public List<Detalle> detalle = new List<Detalle>();
        [XmlElement("Referencia")]
        public List<Referencia> referencia = new List<Referencia>();
    }
    public class Encabezado
    {
        [XmlElement("IdDoc")]
        public IdDoc iddoc = new IdDoc();
        [XmlElement("Emisor")]
        public Emisor emisor = new Emisor();
        [XmlElement("Receptor")]
        public Receptor receptor = new Receptor();
        [XmlElement("Totales")]
        public Totales totales = new Totales();
    }
    public class IdDoc
    {
        [XmlElement("TipoDTE")]
        public int TipoDTE { get; set; }
        [XmlElement("Folio")]
        public int Folio { get; set; }
        [XmlElement("FchEmis")]
        public string FchEmis { get; set; }
        [XmlElement("FmaPago")]
        public int FmaPago { get; set; }
    }
    public class Emisor
    {
        [XmlElement("RUTEmisor")]
        public string RUTEmisor { get; set; }
        [XmlElement("RznSoc")]
        public string RznSoc { get; set; }
        [XmlElement("GiroEmis")]
        public string GiroEmis { get; set; }
        [XmlElement("Acteco")]
        public string Acteco { get; set; }
        [XmlElement("DirOrigen")]
        public string DirOrigen { get; set; }
        [XmlElement("CmnaOrigen")]
        public string CmnaOrigen { get; set; }
        [XmlElement("CiudadOrigen")]
        public string CiudadOrigen { get; set; }
    }
    public class Receptor
    {
        [XmlElement("RUTRecep")]
        public string RUTRecep { get; set; }
        [XmlElement("RznSocRecep")]
        public string RznSocRecep { get; set; }
        [XmlElement("GiroRecep")]
        public string GiroRecep { get; set; }
        [XmlElement("DirRecep")]
        public string DirRecep { get; set; }
        [XmlElement("CmnaRecep")]
        public string CmnaRecep { get; set; }
        [XmlElement("CiudadRecep")]
        public string CiudadRecep { get; set; }
    }
    public class Totales
    {
        [XmlElement("MntNeto")]
        public decimal MntNeto { get; set; }
        [XmlElement("MntExe")]
        public decimal MntExe { get; set; }
        [XmlElement("TasaIVA")]
        public decimal TasaIVA { get; set; }
        [XmlElement("IVA")]
        public decimal IVA { get; set; }
        [XmlElement("MntTotal")]
        public decimal MntTotal { get; set; }
    }
    public class Detalle
    {
        [XmlElement("NroLinDet")]
        public int NroLinDet { get; set; }
        [XmlElement("CdgItem")]
        public List<CdgItem> cdgItem = new List<CdgItem>();
        [XmlElement("NmbItem")]
        public string NmbItem { get; set; }
        [XmlElement("DscItem")]
        public string DscItem { get; set; }
        [XmlElement("QtyItem")]
        public int QtyItem { get; set; }
        [XmlElement("PrcItem")]
        public decimal PrcItem { get; set; }
        [XmlElement("MontoItem")]
        public decimal MontoItem { get; set; }
    }

    public class CdgItem
    {
        [XmlElement("TpoCodigo")]
        public string TpoCodigo { get; set; }
        [XmlElement("VlrCodigo")]
        public string VlrCodigo { get; set; }
    }
    public class Referencia
    {
        [XmlElement("NroLinRef")]
        public int NroLinRef { get; set; }

        [XmlElement("TpoDocRef")]
        public int TpoDocRef { get; set; }

        [XmlElement("FolioRef")]
        public int FolioRef { get; set; }

        [XmlElement("FchRef")]
        public int FchRef { get; set; }

        [XmlElement("CodRef")]
        public int CodRef { get; set; }

        [XmlElement("RazonRef")]
        public int RazonRef { get; set; }
    }

   
    public class Ids
    {
        public int IdEmpresa { get; set; }
        public int IdEmisor { get; set; }
        public int IdReceptor { get; set; }
    }

}

File: app_Factura/Common/App/Utilities/dte.js
$(document).ready(function() {
    dteClass = {
        dte: {
            version: '1.0',
            documento: {
                encabezado: {
                    iddoc: {
                        TipoDTE: 0,
                        Folio: 0,
                        FchEmis: null,
                        FmaPago: 0
                    },
                    emisor: {
                        RUTEmisor: null,
                        RznSoc: null,
                        GiroEmis: null,
                        Acteco: null,
                        DirOrigen: null,
                        CmnaOrigen: null,
                        CiudadOrigen: null
                    },
                    receptor: {
                        RUTRecep: null,
                        RznSocRecep: null,
                        GiroRecep: null,
                        DirRecep: null,
                        CmnaRecep: null,
                        CiudadRecep: null
                    },
                    totales: {
                        MntNeto: 0,
                        MntExe: 0,
                        TasaIVA: 0,
                        IVA: 0,
                        MntTotal: 0
                    }
                },
                ID: null
            },
            detalle: [],
            referencia: []
      
        }
    };

    fnGetJsonDte = function () {
        dteClass = {
            dte: {
                version: '1.0',
                TipoOperacion: '',
                documento: {
                    encabezado: {
                        iddoc: {
                            TipoDTE: 0,
                            Folio: 0,
                            FchEmis: null,
                            FmaPago: 0
                        },
                        emisor: {
                            RUTEmisor: null,
                            RznSoc: null,
                            GiroEmis: null,
                            Acteco: null,
                            DirOrigen: null,
                            CmnaOrigen: null,
                            CiudadOrigen: null
                        },
                        receptor: {
                            RUTRecep: null,
                            RznSocRecep: null,
                            GiroRecep: null,
                            DirRecep: null,
                            CmnaRecep: null,
                            CiudadRecep: null
                        },
                        totales: {
                            MntNeto: 0,
                            MntExe: 0,
                            TasaIVA: 0,
                            IVA: 0,
                            MntTotal: 0
                        }
                    },
                    ID: null,
                    detalle: [],
                    referencia: []
                }

            }
        }

        return dteClass;
    }

    Detalle = {
        NroLinDet: 0,
        NmbItem: null,
        DscItem: null,
        QtyItem: 0,
        PrcItem: 0,
        MontoItem: 0
    }

    var Host = 'http://localhost:57530/ws_Sii.svc';

    fnGetPreviewDte = function(ObjDte, callback) {
        $.ajax({
            type: 'POST',
            url: Host + '/GetPreviewDTE',
            data: JSON.stringify(ObjDte),
            contentType: 'application/json; charset=utf-8',
            dataType: 'json',
            success: function(msg) {
                callback(JSON.parse(msg.d));
            },
            error: function(request, status, error) {
                callback(request.responseText);
            }
        });
    }

    fnSaveDocDte = function (ObjDte, callback) {
        $.ajax({
            type: 'POST',
            url: Host + '/SaveDocDte',
            data: JSON.stringify(ObjDte),
            contentType: 'application/json; charset=utf-8',
            dataType: 'json',
            success: function (msg) {
                callback(JSON.parse(msg.d));
            },
            error: function (request, status, error) {
                callback(request.responseText);
            }
        });
    }

    fnSavePfx = function (ObjCertificadoDigital, callback) {
        $.ajax({
            type: 'POST',
            url: Host + '/SavePfx',
            data: JSON.stringify(ObjCertificadoDigital),
            contentType: 'application/json; charset=utf-8',
            dataType: 'json',
            success: function (msg) {
                callback(JSON.parse(msg.d));
              
            },
            error: function (request, status, error) {
                callback(request.responseText);
            }
        });
    }

    fnAddDetalle = function(dteClass, Detalle) {
        dteClass.dte.detalle.push(Detalle);
    }
});